src/share/classes/com/sun/btrace/runtime/instr/SynchronizedInstrumentor.java

```diff
@@ -74,6 +74,18 @@ public void visitInsn(int opcode) {
         } else if (opcode == MONITOREXIT) {
             onBeforeSyncExit();
         }
+        switch (opcode) {
```

Comment:
I think it would be better to go up the class hierarchy and make `ErrorReturnInstrumentor` to extend `MethodReturnInstrumentor` which it should. Then, instead of repeating the same switch statement you could just call `super.visitInsn(opcode)` and the effect should be the same.